command/connect/envoy/envoy.go

```diff
@@ -887,7 +886,7 @@ func (c *cmd) lookupXDSPort() (int, string, error) {
 
 	var resp response
 	if err := mapstructure.Decode(self, &resp); err == nil {
-		if resp.XDS.Ports.TLS < 0 && resp.XDS.Ports.Plaintext < 0 {
+		if resp.XDS.Ports.TLS <= 0 && resp.XDS.Ports.Plaintext <= 0 {
```

Comment:
I had initially assumed `0` meant "not set by user" whereas `-1` meant "explicitly disabled by user". While this is true during consul agent startup, the response from `/v1/agent/self` actually has an empty XDS field, which decode the port fields to `0`.

This means that `0` effectively means "gRPC is disabled on the agent"